l1-contracts/src/governance/GSE.sol

```diff
@@ -228,43 +338,53 @@ contract GSECore is IGSECore, Ownable {
   {
-    address instanceAddress = msg.sender;
-    InstanceStaking storage instanceStaking = instances[instanceAddress];
-    bool isAttester = instanceStaking.attesters.contains(_attester);
-
-    // If we are canonical, and have not already found the attester we need to look
-    // within the canonical accounting as well - it might be there.
-    // We are figuring out where the attester is effectively located
+    // We need to figure out where the attester is effectively located
+    // we start by looking at the instance that is withdrawing the attester
+    address withdrawingInstance = msg.sender;
+    InstanceStaking storage instanceStaking = instances[msg.sender];
+    bool foundAttester = instanceStaking.attesters.contains(_attester);
+
+    // If we haven't found the attester in the rollup instance staking,
+    // and we are latest rollup, go look in the "bonus" instance.
     if (
-      !isAttester && getCanonical() == instanceAddress
-        && instances[CANONICAL_MAGIC_ADDRESS].attesters.contains(_attester)
+      !foundAttester && getLatestRollup() == msg.sender
+        && instances[BONUS_INSTANCE_ADDRESS].attesters.contains(_attester)
     ) {
-      instanceAddress = CANONICAL_MAGIC_ADDRESS;
-      instanceStaking = instances[instanceAddress];
-      isAttester = true;
+      withdrawingInstance = BONUS_INSTANCE_ADDRESS;
+      instanceStaking = instances[BONUS_INSTANCE_ADDRESS];
+      foundAttester = true;
     }
 
-    require(isAttester, Errors.GSE__NothingToExit(_attester));
+    require(foundAttester, Errors.GSE__NothingToExit(_attester));
 
-    uint256 balance = delegation.getBalanceOf(instanceAddress, _attester);
+    uint256 balance = delegation.getBalanceOf(withdrawingInstance, _attester);
     require(balance >= _amount, Errors.GSE__InsufficientStake(balance, _amount));
 
+    // First assume we are only withdrawing the amount specified.
     uint256 amountWithdrawn = _amount;
+    // If the balance after withdrawal is less than the minimum stake,
+    // we will remove the attester from the instance.
     bool isRemoved = balance - _amount < MINIMUM_STAKE;
 
-    // By default, we will be removing, but in the case of slash, we might just reduce.
+    // Note that the current implementation of the rollup does not allow for partial withdrawals,
+    // via `initiateWithdraw`, so a "normal" withdrawal will always remove the attester from the instance.
+    // However, if the attester is slashed, we might just reduce the balance.
     if (isRemoved) {
       require(instanceStaking.attesters.remove(_attester), Errors.GSE__FailedToRemove(_attester));
       delete instanceStaking.configOf[_attester];
       amountWithdrawn = balance;
 
-      // Update the delegate to address(0) when removing.
-      delegation.delegate(instanceAddress, _attester, address(0));
+      // Update the delegatee to address(0) when removing.
+      // This reduces the voting power of the attester's delegatee by the amount withdrawn.
```

Comment:
The `decreaseBalance` would also handle the decrease in voting power, however we are using this to have a "clean slate" where people with 0 balance should not have a delegatee neither.